doc/Tutorials/Approximate-Synchronizer-Python.rst

```diff
@@ -0,0 +1,155 @@
+---------------------------------------
+
+Prerequisites
+~~~~~~~~~~~~~
+This tutorial assumes you have a working knowledge of ROS 2
+
+If you have not done so already `create a workspace <https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Creating-A-Workspace/Creating-A-Workspace.html>`_ and `create a package <https://docs.ros.org/en/rolling/Tutorials/Beginner-Client-Libraries/Creating-Your-First-ROS2-Package.html>`_
+
+1. Create a Basic Node with Includes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: Python
+
+    import rclpy
+    from rclpy.node import Node
+    from rclpy.qos import QoSProfile
+    from rclpy.clock import Clock
+
+    from message_filters import Subscriber, ApproximateTimeSynchronizer
+    from sensor_msgs.msg import Temperature, FluidPressure
+
+
+    class TimeSyncNode(Node):
+
+        def __init__(self):
+            super().__init__('sync_node')
+            qos = QoSProfile(depth=10)
+            self.temp_pub = self.create_publisher(Temperature, 'temp', qos)
+            self.fluid_pub = self.create_publisher(FluidPressure, 'fluid', qos)
+            self.temp_sub = Subscriber(self, Temperature, "temp")
+            self.fluid_sub = Subscriber(self, FluidPressure, "fluid")
+
+            self.timer = self.create_timer(1, self.TimerCallback)
+            self.second_timer = self.create_timer(1.05, self.SecondTimerCallback)
+
+            queue_size = 10
+            max_delay = 0.05
+            self.time_sync = ApproximateTimeSynchronizer([self.temp_sub, self.fluid_sub],
+                                                         queue_size, max_delay)
+            self.time_sync.registerCallback(self.SyncCallback)
+
+
+For this example we will be using the ``temperature`` and ``fluid_pressure`` messages found in
+`sensor_msgs <https://github.com/ros2/common_interfaces/tree/rolling/sensor_msgs/msg>`_.
+To simulate a working ``ApproximateTimeSynchronizer``. We will be publishing and subscribing to topics of those respective types, to showcase how real sensors would be working. To simulate them we will also need two ``Timers`` on different intervals. Then, we will be utilizing an ``ApproximateTimeSynchronizer`` to get these messages from the sensor topics aligned with a slight delay between messages..
+
+
+It is essential that the QoS is the same for all of the publishers and subscribers, otherwise the Message Filter cannot align the topics together. So, create one ``QoSProfile`` and stick with it, or find out what ``qos`` is being used in the native sensor code, and replicate it. Do basic construction of each object relating to the ``Node`` and callback methods that may be used in the future. Both of the two timers we utilize will have different timer values of ``1`` and ``1.05`` which causes the timers to off at different points, which is an advantage of using ``ApproximateTime``. Notice that we must call ``sync->registerCallback`` to sync up the two (or more) chosen topics.
```

Comment:
```suggestion
To simulate a working ``ApproximateTimeSynchronizer``. We will be publishing and subscribing to topics of those respective types, to showcase how real sensors would be working. To simulate them we will also need two ``Timers`` on different intervals. Then, we will be utilizing an ``ApproximateTimeSynchronizer`` to get these messages from the sensor topics aligned with a slight delay between messages..

It is essential that the QoS is the same for all of the publishers and subscribers, otherwise the Message Filter cannot align the topics together. So, create one ``QoSProfile`` and stick with it, or find out what ``qos`` is being used in the native sensor code, and replicate it. Do basic construction of each object relating to the ``Node`` and callback methods that may be used in the future. Both of the two timers we utilize will have different timer values of ``1`` and ``1.05`` which causes the timers to off at different points, which is an advantage of using ``ApproximateTime``. Notice that we must call ``sync->registerCallback`` to sync up the two (or more) chosen topics.
```